TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/Result.java

```diff
@@ -0,0 +1,35 @@
+package com.novoda.tpbot;
+
+import com.novoda.tpbot.support.Optional;
+
+public class Result {
+
+    private final Optional<String> message;
+    private final Optional<Exception> exception;
+
+    public static Result from(String message) {
+        return new Result(Optional.of(message), Optional.<Exception>absent());
+    }
+
+    public static Result from(Exception exception) {
+        return new Result(Optional.<String>absent(), Optional.of(exception));
+    }
+
+    Result(Optional<String> message, Optional<Exception> exception) {
+        this.message = message;
+        this.exception = exception;
+    }
+
+    public Optional<String> message() {
+        return message;
+    }
+
+    public Optional<Exception> exception() {
+        return exception;
+    }
+
+    public boolean isError() {
+        return exception.isPresent();
```

Comment:
Ok, let's keep this one going, we can see in future PRs if we'll have some special cases